packages/blade/src/components/Input/DropdownInputTriggers/AutoComplete.tsx

```diff
@@ -0,0 +1,198 @@
+    setActiveTagIndex,
+    setActiveIndex,
+    filteredValues: globalFilteredValues,
+    selectionType,
+  } = useDropdown();
+
+  React.useEffect((): void => {
+    const firstItemOptionIndex = options.findIndex(
+      (option) => option.value === globalFilteredValues[0],
+    );
+
+    if (firstItemOptionIndex >= 0) {
+      setActiveIndex(firstItemOptionIndex);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [globalFilteredValues.length]);
+
+  const onInputValueChange: BaseDropdownInputTriggerProps['onInputValueChange'] = ({
+    name,
+    value,
+  }) => {
+    setInputValue(value ?? '');
+    props.onInputValueChange?.({ name, value });
+    setActiveTagIndex(-1);
+
+    if (!isOpen) {
+      setIsOpen(true);
+    }
+
+    if (!props.filteredValues) {
+      // eslint-disable-next-line no-lonely-if
+      if (value && options && options.length > 0) {
+        const filteredOptions = getOptionValues().filter((optionValue) =>
+          optionValue.toLowerCase().startsWith(value.toLowerCase()),
+        );
+        setGlobalFilteredValues(filteredOptions);
+      } else {
+        setGlobalFilteredValues(getOptionValues());
+      }
+    }
+  };
+
+  const onTriggerKeydown: BaseDropdownInputTriggerProps['onTriggerKeydown'] = (e) => {
+    if (e.key === 'Enter') {
+      setActiveTagIndex(-1);
+      setGlobalFilteredValues(getOptionValues());
+    } else if (
+      e.key === 'Backspace' &&
+      !inputValue &&
+      activeTagIndex < 0 &&
+      selectedIndices.length > 0
+    ) {
+      if (isControlled) {
+        setControlledValueIndices(selectedIndices.slice(0, -1));
+      } else {
+        setSelectedIndices(selectedIndices.slice(0, -1));
+      }
+    }
+    onBaseDropdownInputKeydown?.(e);
+  };
+
+  const onSelectionChange: BaseDropdownInputTriggerProps['onChange'] = ({ values }) => {
+    if (selectionType === 'multiple') {
+      setInputValue('');
+      props.onInputValueChange?.({ name: props.name, value: '' });
+      setActiveTagIndex(-1);
+      setGlobalFilteredValues(getOptionValues());
+    } else {
+      const displayText = options.find((option) => option.value === values[0])?.title;
+      props.onInputValueChange?.({
+        name: props.name,
+        value: displayText,
+      });
+      setInputValue(displayText ?? '');
+    }
+    props.onChange?.({ name: props.name, values });
+  };
+
+  return {
+    onSelectionChange,
+    onTriggerKeydown,
+    onInputValueChange,
+  };
+};
+
+const _AutoComplete = (
+  props: AutoCompleteProps,
+  ref: React.ForwardedRef<BladeElementRef>,
+): React.ReactElement => {
+  const [uncontrolledInputValue, setInputValue] = React.useState('');
+  const inputValue = props.inputValue ?? uncontrolledInputValue;
+
+  const {
+    isOpen,
+    options,
+    setFilteredValues: setGlobalFilteredValues,
+    hasAutoCompleteInBottomSheetHeader,
+    setHasAutoCompleteInBottomSheetHeader,
+    onTriggerClick,
+    dropdownTriggerer,
+  } = useDropdown();
```

Comment:
This would fail if used outside dropdown context, right? I don't see `useDropdown` doing a context check either?

Reply:
It actually works outside dropdown context. It just won't be able to do filtering etc. When used outside dropdown context, it renders like a simple list.